### PR TITLE
RHOAIENG-11274: chore(odh/notebooks): widen the container build context to the whole repo, so we can start reusing code between images in the future

### PR DESCRIPTION
**Pull request by jiridanek**
Relates to

* https://github.com/opendatahub-io/notebooks/pull/792

Also opened this one to ease testing; this is according to advice from testplatform slack channel

* https://github.com/openshift/release/pull/59534

### CODE REVIEW

**Comment by jiridanek** [comment]
the rehearse cannot pass until the repo itself is also restructured (the related PR linked on top), so I'm acking rehearsals

/pj-rehearse ack

File: ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml
Comment (jiridanek) [inline ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:1]: /testwith jiridanek/release/jd_change_context/notebook-codeserver-ubi9-python-3-11-pr-image-mirror

**Comment by atheo89** [comment]
/testwith jiridanek/release/jd_change_context/notebook-codeserver-ubi9-python-3-11-pr-image-mirror

**Comment by atheo89** [comment]
/testwith jiridanek/release/jd_change_context/ci/prow/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith jiridanek/release/jd_change_context/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/jd_change_context/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror https://github.com/opendatahub-io/notebooks/pull/792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror https://github.com/opendatahub-io/notebooks/pull/792

**Comment by jiridanek** [comment]
/testwith openshift/release/master/notebook-cuda-jupyter-min-ubi9-python-3-11-pr-image-mirror https://github.com/opendatahub-io/notebooks/pull/792 openshift/release#59520

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror opendatahub-io/notebooks#792

**Comment by jiridanek** [comment]
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-9-pr-image-mirror opendatahub-io/notebooks#792
/testwith opendatahub-io/notebooks/main/runtime-minimal-ubi9-python-3-11-pr-image-mirror opendatahub-io/notebooks#792

**Pull request merged**